packages/core/src/fields/resolve-hooks.ts

```diff
@@ -11,68 +23,47 @@ function splitValidateHooks <ListTypeInfo extends BaseListTypeInfo> (
     }
   }
 
-  if (!validate) return undefined
-
+  if (!validate) return
   if (typeof validate === 'function') {
-    return { create: validate, update: validate, delete: validate }
+    return {
+      create: validate,
+      update: validate,
+      delete: validate
+    }
   }
 
   return validate
 }
 
-// force new syntax for built-in fields
-// also, we don't allow built-in hooks to specify resolveInput,
-// since they can do it via graphql resolvers
-export type InternalFieldHooks<ListTypeInfo extends BaseListTypeInfo> =
-  Omit<FieldHooks<ListTypeInfo>, 'validateInput' | 'validateDelete' | 'resolveInput'>
+function merge <
+  R,
+  A extends (r: R) => MaybePromise<void>,
+  B extends (r: R) => MaybePromise<void>
+> (a?: A, b?: B) {
+  if (!a && !b) return undefined
+  return async (args: R) => {
+    await a?.(args)
+    await b?.(args)
+  }
+}
 
-/**
- * Utility function to convert deprecated field hook syntax to the new syntax
- * Handles merging any built-in field hooks into the user-provided hooks
- */
 export function mergeFieldHooks <ListTypeInfo extends BaseListTypeInfo> (
   builtin?: InternalFieldHooks<ListTypeInfo>,
-  hooks: FieldHooks<ListTypeInfo> = {},
-): FieldHooks<ListTypeInfo> {
+  hooks?: FieldHooks<ListTypeInfo>,
+) {
+  if (hooks === undefined) return builtin
   if (builtin === undefined) return hooks
 
-  const result: FieldHooks<ListTypeInfo> = {
-    resolveInput: hooks?.resolveInput,
-  }
-
-  if (hooks.beforeOperation || builtin.beforeOperation) {
-    result.beforeOperation = async (args) => {
-      await hooks.beforeOperation?.(args)
-      await builtin.beforeOperation?.(args)
```

Comment:
@acburdine quick question, did you have any intent behind `hooks.beforeOperation` being first?